CovidWatch/app/src/main/java/org/covidwatch/android/ble/BLEForegroundService.kt

```diff
@@ -61,33 +116,22 @@ class BLEForegroundService : LifecycleService() {
         timer?.scheduleAtFixedRate(
             object : TimerTask() {
                 override fun run() {
-                    app?.bleAdvertiser?.changeContactEventIdentifierInServiceDataField()
+                    app?.cenAdvertiser?.updateCEN()
                 }
             },
             TimeUnit.MINUTES.toMillis(BluetoothService.CONTACT_EVENT_NUMBER_CHANGE_INTERVAL_MIN.toLong()),
             TimeUnit.MINUTES.toMillis(BluetoothService.CONTACT_EVENT_NUMBER_CHANGE_INTERVAL_MIN.toLong())
         )
 
-        val newContactEventUUID = UUID.randomUUID()
-        CovidWatchDatabase.databaseWriteExecutor.execute {
-            val dao: ContactEventDAO = CovidWatchDatabase.getInstance(this).contactEventDAO()
-            val contactEvent = ContactEvent(newContactEventUUID.toString())
-            val isCurrentUserSick = getSharedPreferences(
-                getString(R.string.preference_file_key),
-                Context.MODE_PRIVATE
-            ).getBoolean(getString(R.string.preference_is_current_user_sick), false)
-            contactEvent.wasPotentiallyInfectious = isCurrentUserSick
-            dao.insert(contactEvent)
-        }
-        app?.bleAdvertiser?.startAdvertising(BluetoothService.CONTACT_EVENT_SERVICE, newContactEventUUID)
-        app?.bleScanner?.startScanning(arrayOf<UUID>(BluetoothService.CONTACT_EVENT_SERVICE))
-
+        // start contact logging
+        app!!.cenAdvertiser!!.startAdvertiser(BluetoothService.CONTACT_EVENT_SERVICE)
```

Comment:
Using `!!` is bad practice in Kotlin. You could just replace all of them with `?` and it will be safier.